src/os/ui/draw/drawcontrols.js

```diff
@@ -57,16 +68,77 @@ class Controller extends BaseDrawControlsCtrl {
 
     // Base draw controller doesn't support lines as a default.
     this['supportsLines'] = true;
-
     this.log = logger;
+
+    /**
+     * Flag for whether we are currently measuring.
+     * @type {boolean}
+     * @protected
+     */
+    this.measuring = false;
+
+    /**
+     * @type {!string}
+     * @private
+     */
+    this.key_ = 'measureMethod';
+    Measure.method = /** @type {Method} */ (Settings.getInstance().get(this.key_, Measure.method));
+
+    dispatcher.getInstance().listen(DrawEventType.DRAWEND, this.onDrawEnd_, false, this);
+    dispatcher.getInstance().listen(DrawEventType.DRAWCANCEL, this.onDrawEnd_, false, this);
   }
 
   /**
    * @inheritDoc
    */
   $onDestroy() {
-    super.$onDestroy();
     googEvents.unlisten(getIMapContainer(), MapEvent.MAP_READY, this.onMapReady, false, this);
+    dispatcher.getInstance().unlisten(DrawEventType.DRAWEND, this.onDrawEnd_, false, this);
+    dispatcher.getInstance().unlisten(DrawEventType.DRAWCANCEL, this.onDrawEnd_, false, this);
+    this.controlMenu.unlisten(Method.GEODESIC, this.onMeasureTypeChange_, false, this);
+    this.controlMenu.unlisten(Method.RHUMB, this.onMeasureTypeChange_, false, this);
+    super.$onDestroy();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  initControlMenu() {
+    super.initControlMenu();
+
+    var mi = this.controlMenu.getRoot();
+    mi.addChild({
+      type: MenuItemType.SEPARATOR,
```

Comment:
Should these use `MenuItemType.GROUP`? I'm not sure what the difference is here between that and using a separator with a label.

Reply:
Separators were originally only rendered as a horizontal line. I changed the rendering to support showing them as a sorted header with no underlying hierarchical structure. This was more convenient for cases where I didn't want to refactor the menu to add stuff to a child group, I just wanted a separator with some text on it.